discovery-provider/src/tasks/cache_user_balance.py

```diff
@@ -0,0 +1,148 @@
+import logging
+import time
+from src import eth_abi_values
+from src.tasks.celery_app import celery
+from src.models import UserBalance, User
+from src.queries.get_balances import does_user_balance_need_refresh, REDIS_PREFIX
+
+logger = logging.getLogger(__name__)
+audius_token_registry_key = bytes("Token", "utf-8")
+
+# *Explanation of user balance caching*
+# In an effort to minimize eth calls, we look up users embedded in track metadata once per user,
+# and current users (logged in dapp users, who might be changing their balance) on an interval.
+#
+# - In populate_track_metadata, look up User_Balance entry in db.
+#       If it doesn't exist, return 0, persist a User_Balance row with 0,
+#       & enqueue in Redis the user ID for later balance lookup.
+# - On track get endpoints, if current_user exists, enqueue balance lookup
```

Comment:
comment should be populate_user_metadata right?

couldn't we also stick the current_user_id thing in that function too? it seems like maybe we would want to enqueue the current user in requests for users too? (in practice probs no diff tho)

Reply:
yeah you right, it's populate_user_metadata

in practice no difference, but for simplicity i'm down to just keep it all in populate_user_metadata